subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AnnotationProcessingCompileTask.java

```diff
@@ -96,17 +97,20 @@ public Boolean call() {
 
     private void setupProcessors() {
         processorClassloader = createProcessorClassLoader();
+        SupportedOptionsCollectingProcessor supportedOptionsCollectingProcessor = new SupportedOptionsCollectingProcessor();
         List<Processor> processors = new ArrayList<Processor>(processorDeclarations.size());
         for (AnnotationProcessorDeclaration declaredProcessor : processorDeclarations) {
             AnnotationProcessorResult processorResult = new AnnotationProcessorResult(result, declaredProcessor.getClassName());
             result.getAnnotationProcessorResults().add(processorResult);
 
             Class<?> processorClass = loadProcessor(declaredProcessor);
             Processor processor = instantiateProcessor(processorClass);
+            supportedOptionsCollectingProcessor.addProcessor(processor);
             processor = decorateForIncrementalProcessing(processor, declaredProcessor.getType(), processorResult);
             processor = decorateForTimeTracking(processor, processorResult);
             processors.add(processor);
         }
+        processors.add(supportedOptionsCollectingProcessor);
```

Comment:
❌ Should only add this if there are processors, otherwise we're triggering annotation processing code for no reason.

Reply:
Ah yes, I was thinking about that a few hours ago (but had switched to something else and haven't checked back yet), wondering whether this class would be used every time, or only when processors are present.

Added the check here to avoid unnecessarily triggering processing.